tests/neptune/test_base.py

```diff
@@ -64,6 +64,80 @@ def test_delete_atom(self, run):
             run[key].fetch()
 
 
+class TestNamespace(BaseE2ETest):
```

Comment:
tbh I was thinking about fewer-but-bigger e2e tests. Less granularity, e.g. one-two tests for whole "assigning and fetching namespaces usecase". 

Or maybe even "user scenarios" instead of synthetic "test a specific function"?

Putting it up for discussion.

Reply:
I like the idea of a test that mimics a user, but I believe that the best approach for us devs is to have one "check it all" test and more specialized tests that basically do one thing.

From experience, such approach will make it easier for us to pinpoint what's wrong with what functianality.

Making one big test and having it fail often would encourage `it's broken anyway` and mentally ignoring that it fails all the time.

Another thing with `one big test` approach is, if something early is broken, then we will not see further issues unless we fix the early error

Reply:
Personally I think that one `TestUnit` should be as small as possible, and shouldn't have a lot of functions.

I think that we should have bigger chunks of tests, such as particular integrations or user scenarios, but I think that those groups should be managed on `package` level.

Reply:
Actually we always should think about "what regression can be made here and what will be an impact on user-side". I've rewritten tests structure to cover two cases:
- User works with Namespaces: assign internal keys, updates 
- User done some "unsupported behavior": provides distinct (in context of type casting) type during an update of metadata

I don't think it should 1-1 cover every "public endpoint method" with(/-out) a context of user scenarios but just keep in mind a SRP principle :wink: